### ISSUE TITLE
private dockerhub registry

### ISSUE THREAD

**Issue by ltetrel** [neurolibre/neurolibre-binderhub#2]
Currently we have networking issues with computecanada. The pulling/building of the docker images takes forever.
To address this, we can create our own docker registry (to address our networking issues)

I know that computecanada is working on that so we should be in touch with them.

**Comment by agahkarakuzu** [comment]
Longevity of the images pushed to the registry is also important. For example, I don't know what's gonna happen to my images on DockerHub in 3 years. Looking from this window, I think the following question becomes relevant: 

Do we want 100% control over managing user images OR can we count on someone else hosting them (even if we are talking about a non-neurolibre CC entity).

**Comment by ltetrel** [comment]
I don't think we want to host it somewhere else than in CC server. We would have no benefits with that since we would be also dependent from the internet speed issue. Again our internet speed throttle down on CC side (and depending on their demand), it is not the fault of Docker (which I think have very strong infrastructure to serve all the users around the world).

**Comment by agahkarakuzu** [comment]
> I don't think we want to host it somewhere else than in CC server.

I did not mean to host it elsewhere. My point was about the "ownership". Ofc it has nothing to do with central Docker registry if our transfer bw is low.

**Comment by ltetrel** [comment]
For more information on how to set up a private registry on binderhub :
https://binderhub.readthedocs.io/en/latest/setup-binderhub.html#if-you-are-using-a-custom-registry
